src/app/calendar/parcelCalendarFunctions.ts

```diff
@@ -0,0 +1,46 @@
+import supabase, { Schema } from "@/supabase";
+import { CalendarEvent } from "@/components/Calendar/Calendar";
+
+export type parcelWithClientName = Schema["parcels"] & { clients: { full_name: string } | null };
+
+export interface LocationColorMap {
+    [location: string]: { color: string; text: string };
+}
+
+const COLLECTION_DURATION_MS = 30 * 60 * 1000;
+
+export const getParcelsWithCollectionDate = async (): Promise<parcelWithClientName[]> => {
+    const response = await supabase
+        .from("parcels")
+        .select("*, clients ( full_name )")
+        .not("collection_datetime", "is", null);
+    return response.data ?? [];
```

Comment:
If there is no response do we want some error to be thrown instead of just an empty array? Or is this saying if there is no data for that day return an empty array. In any case, there should be some way for the user to differentiate between the two occurences.

Reply:
This is a decision made by the previous calendar commit 
I think we haven't had an error page properly configured so that's why we are doing this now

Reply:
I feel like we should still have some way of differentiating between the two though.